### DISCUSSION TITLE
Variable naming

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#59] [Notes (internal)]
When deciding new variable names, I'd like to lean on names that other plugins, or Obsidian itself, already use. For example, `{{date:format}}` is the same that Obsidian's *daily note* core plugin uses in template files, and even the *format* part uses exactly the same syntax. I want this so that variables would be as seamless as possible to use for users when they use different parts of their Obsidian setup.

I have to admit that I'm not so familiar with variables in other plugins. I created this topic so that this could contain a list of variable names that are used in core plugins and in community plugins. No matter what the variable is about or in which kind of plugin it's used in. I'd just like to have a comprehensive list.

You can conribute to this list by writing down a plugin name (with link to it's Github repository) and all of its variable names with short descriptions. Thank you very much, your help is appreciated!

## Obsidian core plugins

| Variable name     | Description                                             | Core plugin              |
| ----------------- | ------------------------------------------------------- | ------------------------ |
| `{{content}}`     | Extracted note content.                                 | Note composer            |
| `{{date:FORMAT}}` | Current date.                                           | Note composer, Templates |
| `{{fromTitle}}`   | The name of the note where an extracted note comes from | Note composer            |
| `{{newTitle}}`    | The title of the new file after extraction.             | Note composer            |
| `{{time}}`        | Current time.                                           | Templates                |
| `{{title}}`       | Current file's name without .md extension.              | Templates                |

### Search operators

| Search operator | Description                                                                                                                                      |
| --------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
| `block:`        | Performs the subquery on a block-by-block basis, where each block defined as a markdown block, typically separated by empty lines or list items. |
| `content:`      | Will search for a file content.                                                                                                                  |
| `file:`         | Will search for a file name.                                                                                                                     |
| `ignore-case:`  |                                                                                                                                                  |
| `line:`         | Performs the subquery on a line-by-line basis, rather than a file-by-file basis.                                                                 |
| `match-case:`   |                                                                                                                                                  |
| `path:`         |                                                                                                                                                  |
| `section:`      | Performs the subquery on a section-by-section basis, where each section is the text between two headings, including the first heading.           |
| `tag:`          |                                                                                                                                                  |
| `task:`         | Will search for tasks.                                                                                                                           |
| `task-todo:`    |                                                                                                                                                  |
| `task-done:`    |                                                                                                                                                  |

### [Snippets plugin](https://github.com/cristianvasquez/obsidian-snippets-plugin)

This plugin happens to have many similar variable names as *Shell commands* (except `{{folder}}`).

| Variable name    | Description |
| ---------------- | ----------- |
| `{{file_name}}`  |             |
| `{{file_path}}`  |             |
| `{{folder}}`     | Folder name |
| `{{vault_path}}` |             | 

P.S. Now that I'm posting in the _Notes_ category, I know that it's a locked category where only I can create new posts, but can for example @FelipeRearden test to write a comment here just so that I can see that others can comment here, please? If you can't comment here, please mention about it in some other discussion. Thanks! 🙂

**Comment by FelipeRearden** [comment]
### Obsidian Core
- Row = Line
- Content of a File = Note
- A window in the screen that is used to open Notes = Pane (open in a new pane)

**Reply by FelipeRearden** [reply]
I think I can create comments :)

**Reply by Taitava** [reply]
Nice! 🙂

**Reply by Taitava** [reply]
Where are these *Note* and *Line* used in Obsidian? Are there somewhere `{{note}}` or `{{line}}`?

I found [*line:* in Obsidian's Help vault](https://help.obsidian.md/Plugins/Search#Search+operators), but not *note*. But this brings to my mind, that I could also list Obsidian's search parameters in this topic. They are not actually variables, but they could offer good naming convention ideas for variables.

**Reply by FelipeRearden** [reply]
> They are not actually variables, but they could offer good naming convention ideas for variables.

That’s exactly what I was trying to say when I wrote Note and Line !

We can use name convention of Obsidian as words for variables names for Shell Commands.
